test/e2e/framework/fixture.go

```diff
@@ -185,3 +202,51 @@ func NewWorkspaceFixture(t *testing.T, server RunningServer, orgClusterName stri
 
 	return helper.EncodeOrganizationAndWorkspace(orgName, ws.Name)
 }
+
+type ControllerFixture interface {
+	// GetName will return the name of the controller. This can be
+	// used in cases where the fixture needs to be identified for
+	// logging purposes.
+	GetName() string
+	// GetServerArgs returns the cli arguments required to configure
+	// kcp to run the controller. This will be called when configuring
+	// an out-of-process kcp server managed by the test run.
+	GetServerArgs() []string
+	// Configure enables the controller fixture to run itself against
+	// the given server. This will be called when running the
+	// controller in-process.
+	Run(t *testing.T, server RunningServer)
+}
+
+type ClusterControllerFixture struct {
+	Args                []string
+	InProcessServerArgs []string
+}
+
+func (f *ClusterControllerFixture) GetName() string {
+	return "ClusterController"
+}
+
+func (f *ClusterControllerFixture) GetServerArgs() []string {
+	if TestConfig.InProcessControllers {
+		return append(f.Args, f.InProcessServerArgs...)
+	}
+	return f.Args
+}
+
+func (f *ClusterControllerFixture) Run(t *testing.T, server RunningServer) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
+	f.Args = append(f.Args, fmt.Sprintf("--kubeconfig=%s", server.KubeconfigPath()))
```

Comment:
@sttts From what I can tell, this kubeconfig is no longer sufficient to configure the cluster controller. The cluster controller test fails trying to install the cluster due to the cluster never becoming ready. When I debugged the in-process sync manager controller, it wasn't receiving events at all. 

Is there something about the configuration here that differs from what is supplied to the cluster controller installed by the kcp server?